File: ccc20s3.cpp
#include <stdio.h>
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

int main() {
	int SIZE = 200000;
	char needleString[SIZE];
	char haystackString[SIZE];
	scanf("%s", needleString);
	scanf("%s", haystackString);
	sort(needleString, needleString + SIZE);
	vector<string> tempArray = {};
	int c = 0;
	for (int i = 0; i < haystackString.size() - (needleString.size() - 1); i++) {
		/*
		for (int j = 0; j < needle.length; j++) {
			tempArray[i] = haystack[i + j];
		}
		
		if (! checked.contains(new String(tempArray))) {
			checked.add();
			if (Arrays.equals(tempArray, needle)) {
				c++;
			}
		}
		*/
	}
	cout << c;
}

File: ccc20s3.java
import java.util.*;

public class ccc20s3 {
	public static void main(String[] args) {
		Scanner input = new Scanner(System.in);
		char[] needleArray = input.nextLine().toCharArray();
		Arrays.sort(needleArray);
		String needle = new String(needleArray);
		
		String haystack = input.nextLine();
		List<String> list = new ArrayList();
		char[] tempArray;
		
		int c = 0;
		for (int i = 0; i < haystack.length() - (needle.length() - 1); i++) {
			String temp = haystack.substring(i, i + needle.length());
			if (! list.contains(temp)) {
				list.add(temp);
				tempArray = temp.toCharArray();
				Arrays.sort(tempArray);
				if (new String(tempArray).equals(needle)) {
					c++;
				}
			}
		}
		System.out.println(c);
	}
}
